clang/lib/Sema/SemaTemplateInstantiateDecl.cpp

```diff
@@ -8,6 +8,8 @@
 //  This file implements C++ template instantiation for declarations.
 //
 //===----------------------------------------------------------------------===/
+#include "SemaTemplateImpl.h"
```

Comment:
Are template methods from new header are used in this source file?
If not, please, remove this include.

Reply:
Alexey, they are.   The first one - AddOneConstantValueAttr - is used at around lines 528 and 539.   The second one - AddOneConstantPowerTwoValueAttr - is used at around lines 506 and 517.

Reply:
How is this working today if these are defined in other source file?
Are we instantiating the same specializations in both files?
How did this code pass code review?

Reply:
It works today because when the files are linked together, the definition is found from the other file.  The declaration is seen from Sema.h, so it compiles fine.   The reason the issue was found is that we have a more optimizing build where the template definition got completely inlined, and the other file couldn't find the definition and failed at link-time.

Yes, we are instantiating the same specialization in both files.

Reply:
Why not define these methods in  `Sema.h` then?

Reply:
I don't think there is a technical reason.   I would rather not have implementations there to reduce the clutter in that file.

Do you feel strongly about this?

Reply:
I think it would be hard to debug cases when Sema.h is included and methods defined in a separate header is used.
Reducing clutter in Sema.h shouldn't be resolved by this PR. Let's handle it separately.